evalml/automl/automl_algorithm/evalml_algorithm.py

```diff
@@ -0,0 +1,367 @@
+import inspect
+
+import numpy as np
+from skopt.space import Categorical, Integer, Real
+
+from .automl_algorithm import AutoMLAlgorithm
+
+from evalml.automl.utils import get_hyperparameter_ranges
+from evalml.model_family import ModelFamily
+from evalml.pipelines.components import (
+    RFClassifierSelectFromModel,
+    RFRegressorSelectFromModel,
+)
+from evalml.pipelines.components.transformers.column_selectors import (
+    SelectColumns,
+)
+from evalml.pipelines.components.utils import (
+    get_estimators,
+    handle_component_class,
+)
+from evalml.pipelines.utils import (
+    _make_stacked_ensemble_pipeline,
+    make_pipeline,
+)
+from evalml.problem_types import is_regression
+
+
+class EvalMLAlgorithm(AutoMLAlgorithm):
+    """An automl algorithm that consists of two modes: fast and long. Where fast is a subset of long.
+
+    1. Naive pipelines:
+        a. run baseline with default preprocessing pipeline
+        b. run naive linear model with default preprocessing pipeline
+        c. run basic RF pipeline (for feature selection) with default preprocessing pipeline
+    2. Feature engineering and naive pipelines with feature selection:
+        a. create feature selection component with previous batches’ RF estimator then add to another linear model
+        b. Run feature engineering: leveraging featuretools and our DFSTransformer
+    3. Naive pipelines with feature engineering
+        a. Use FT component from previous batch with naive linear model and RF pipeline
+    4. Naive pipelines with feature engineering and feature selection
+        a. use previous RF estimator to run FS with naive linear model
+
+    At this point we have a single pipeline candidate for preprocessing, feature engineering and feature selection
+
+    5. Pipelines with preprocessing components:
+        a. scan estimators (our current batch 1).
+        b. Then run ensembling
+
+    Fast mode ends here. Begin long mode.
+
+    6. Run some random pipelines:
+        a. Choose top 3 estimators. Generate 50 random parameter sets. Run all 150 in one batch
+    7. Run ensembling
+    8. Repeat these indefinitely until stopping criterion is met:
+        a. For each of the previous top 3 estimators, sample 10 parameters from the tuner. Run all 30 in one batch
+        b. Run ensembling
+    """
+
+    def __init__(
+        self,
+        X,
+        y,
+        problem_type,
+        _sampler_name,
+        tuner_class=None,
+        random_seed=0,
+        pipeline_params=None,
+        custom_hyperparameters=None,
+        n_jobs=-1,
+        number_features=None,
```

Comment:
Not sure if `number_features` is still necessary anymore. I still see it in `IterativeAlgorithm` but don't see it mentioned anywhere else in the codebase.